crates/pcsc-lite-rs/build.rs

```diff
@@ -0,0 +1,18 @@
+use std::env;
+
+use pkg_config::Config;
+
+fn main() {
+    if let Ok(lib_dir) = env::var("PCSC_LIB_DIR") {
+        println!("cargo:rustc-link-search=native={}", lib_dir);
+        println!(
+            "cargo:rustc-link-lib={}",
+            env::var("PCSC_LIB_NAME").unwrap_or_else(|_| "pcsclite".to_string())
+        );
+    } else {
+        Config::new()
+            .atleast_version("1")
+            .probe("libpcsclite")
+            .expect("Could not find a PCSC library");
+    }
```

Comment:
To clarify -- will this library be statically linked to the resulting executable?

Reply:
Nope, it'll be dynamically linked.
The `pcsc-lite` API is stable and likely will not be changed. So, I decided to link it dynamically

Reply:
Hm... I have a concern that when we start to use this in IronRDP or other products, 90% of the time this library will be absent on the system, and the application will simply fail to start because of that, even if the user does not need smart card auth support at all.

Maybe it is better to use `dlopen`/`dlsym` machinery for smart card auth to load `pcsc-lite` on runtime? Not only this will make `pcsc-lite` presence on the machine optional, but this will also simplify the compilation process of sspi-rs (no need for installing/compiling `pcsc-lite` on dev machine)

Alternatively, we could just statically link the library.

cc @CBenoit @awakecoding

Reply:
> I wonder If we actually had any problems I described above 🤔

we used the `pcsc` crate only on Windows. So, the WinSCard API was always with us on Windows 😅

Reply:
> Maybe it is better to use dlopen/dlsym machinery for smart card auth to load pcsc-lite on runtime? Not only this will make pcsc-lite presence on the machine optional, but this will also simplify the compilation process of sspi-rs (no need for installing/compiling pcsc-lite on dev machine)

static linking is not really an option for pcsc-lite, and yes, adding a runtime dependency that is likely to be missing is a concern, especially on Linux. macOS should not be much of an issue issue though, even if I'd prefer having runtime loading of the DLL. Maybe we can begin by enabling the feature at build time only for macOS, and then rework this to find and load the pcsc-lite library at runtime in a second time before trying to use it on Linux?

Reply:
> even if I'd prefer having runtime loading of the DLL

got it. I'll update the PR tomorrow.
I already implemented a similar behavior during the dev testing. Thus it's not a big problem to add it now

Reply:
> I'll update the PR tomorrow

done